curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md

```diff
@@ -28,26 +28,38 @@ tests:
         assert(!/.*\/anonymous-message-board\.freecodecamp\.rocks/.test(getUserInput('url')));
       }
   - text: Only allow your site to be loading in an iFrame on your own pages.
-    testString: ''
+    testString: "async getUserInput => {
+      const data = await fetch(getUserInput('url') + '/_api/app-info');
+      const parsed = await data.json();
+      assert.isTrue(parsed.headers['x-frame-options']?.includes('SAMEORIGIN'));
+      }"
   - text: Do not allow DNS prefetching.
-    testString: ''
+    testString: "async getUserInput => {
+      const data = await fetch(getUserInput('url') + '/_api/app-info');
+      const parsed = await data.json();
+      assert.isTrue(parsed.headers['x-dns-prefetch-control']?.includes('off'));
+      }"
   - text: Only allow your site to send the referrer for your own pages.
+    testString: "async getUserInput => {
+      const data = await fetch(getUserInput('url') + '/_api/app-info');
+      const parsed = await data.json();
+      assert.isTrue(parsed.headers['referrer-policy']?.includes('same-origin'));
+      }"
+  - text: I can POST a thread to a specific message board by passing form data `text` and `delete_password` to `/api/threads/{board}`.(Recommend `res.redirect` to board page `/b/{board}`) Saved will be at least `_id`, `text`, `created_on`(date & time), `bumped_on`(date & time, starts same as `created_on`), `reported` (boolean), `delete_password`, & `replies` (array).
     testString: ''
-  - text: I can POST a thread to a specific message board by passing form data text and deletepassword_ to /api/threads/{board}.(Recommend res.redirect to board page /b/{board}) Saved will be at least _id, text, createdon_(date&time), bumpedon_(date&time, starts same as created_on), reported(boolean), deletepassword_, & replies(array).
-    testString: ''
-  - text: I can POST a reply to a thread on a specific board by passing form data text, deletepassword_, & threadid_ to /api/replies/{board} and it will also update the bumped_on date to the comments date.(Recommend res.redirect to thread page /b/{board}/{thread_id}) In the thread's replies array will be saved _id, text, createdon_, deletepassword_, & reported.
+  - text: I can POST a reply to a thread on a specific board by passing form data `text`, `delete_password`, & `thread_id` to `/api/replies/{board}` and it will also update the `bumped_on` date to the comment's date.(Recommend `res.redirect` to thread page `/b/{board}/{thread_id}`) In the thread's `replies` array will be saved `_id`, `text`, `created_on`, `delete_password`, & `reported`.
```

Comment:
```suggestion
  - text: I can POST a reply to a thread on a specific board by passing form data `text`, `delete_password`, & `thread_id` to `/api/replies/{board}` and it will also update the `bumped_on` date to the comment's date (Recommend `res.redirect` to thread page `/b/{board}/{thread_id}`). In the thread's `replies` array, an object will be saved with at least the following properties:  `_id`, `text`, `created_on`, `delete_password`, & `reported`.
```